static/app/components/forms/controls/selectControl.tsx

```diff
@@ -115,6 +115,10 @@ export interface ControlProps<OptionType extends OptionTypeBase = GeneralSelectV
    * Set to true to prefix selected values with content
    */
   inFieldLabel?: string;
+  /**
+   * Whether this selector is being rendered inside a modal. If true, the menu will have a higher z-index.
```

Comment:
Does it always portal then?

Reply:
Nope. You still have to set `menuPortalTarget={}`.